pkg/controllers/provisioning/suite_test.go

```diff
@@ -134,33 +133,81 @@ var _ = Describe("Provisioning", func() {
 				ExpectScheduled(ctx, env.Client, pod)
 			}
 		})
-		It("should account for daemonsets", func() {
-			ExpectCreated(env.Client, &appsv1.DaemonSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "daemons", Namespace: "default"},
-				Spec: appsv1.DaemonSetSpec{
-					Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "test"}},
-					Template: v1.PodTemplateSpec{
-						ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "test"}},
-						Spec: test.UnschedulablePod(test.PodOptions{
-							ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
-						}).Spec,
+		Context("Daemonsets and Node Overhead", func() {
+			It("should account for overhead", func() {
+				ExpectCreated(env.Client, test.DaemonSet(
+					test.DaemonSetOptions{PodOptions: test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
 					}},
-			})
-			for _, pod := range ExpectProvisioned(ctx, env.Client, scheduler, controller, provisioner,
-				test.UnschedulablePod(test.PodOptions{
-					ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
-				}),
-				test.UnschedulablePod(test.PodOptions{
-					ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
-				}),
-				test.UnschedulablePod(test.PodOptions{
-					ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
-				}),
-			) {
+				))
+				pod := ExpectProvisioned(ctx, env.Client, scheduler, controller, provisioner, test.UnschedulablePod(
+					test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
+					},
+				))[0]
 				node := ExpectScheduled(ctx, env.Client, pod)
 				Expect(*node.Status.Allocatable.Cpu()).To(Equal(resource.MustParse("4")))
 				Expect(*node.Status.Allocatable.Memory()).To(Equal(resource.MustParse("4Gi")))
-			}
+			})
+			It("should not schedule if overhead is too large", func() {
```

Comment:
Is the reason why this test works that none of the instance types in the fake EC2 implementation have 10Gi of memory? I'd probably at least add a comment to that effect, especially considering this is the main test that tests out that functionality. Also, why not make the daemonset require way more memory and 10Gi just to make it more clear?

Reply:
Good call.